### PR TITLE
Prevent browser from opening on ctrl+c on windows

### PR DESCRIPTION
**Pull request by calufornia**
## Description

Exiting `airplane dev` currently causes the browser to open in Windows. This is because the interrupt signal is getting captured by `fmt.Scanln()`. Furthermore, we always advance past `fmt.Scanln()` _before_ the stop channel receives a notification of an interrupt signal. This PR adds an artificial timeout that allows the stop channel to get notified of the interrupt signal before continuing on to open the browser.

cc @yolken-airplane 

Resolves ASM-20

## Test plan
Verified that the browser-opening issue no longer occurs on windows.

### CODE REVIEW

File: cmd/airplane/tasks/dev/server.go [hunk 1]
```diff
@@ -264,6 +265,17 @@ func runLocalDevServer(ctx context.Context, cfg taskDevConfig) error {
...
 			fmt.Scanln()
+			if runtime.GOOS == "windows" {
+				// This is a hack so that Ctrl+C on Windows doesn't cause the browser to open. Ctrl+C is being captured
+				// by the fmt.Scanln() above, and furthermore it is being captured _before_ the stop channel gets
+				// notified of a termination signal. We add an artificial timeout here to give the stop channel a chance
+				// to get notified first.
+				select {
+				case <-stop:
+					return
+				case <-time.After(100 * time.Millisecond):
```
Comment (calufornia) [inline cmd/airplane/tasks/dev/server.go:276]: This is quite hacky, but consistently prevents the browser from opening on exit and also does not noticeably increase the time it takes between the user pressing enter and the browser opening.

**Comment by calufornia** [comment]
I'm not entirely sure why `fmt.Scanln` is capturing ctrl+c—I think somewhere, the windows interrupt signal is getting converted into a new line. Using bufio.NewScanner doesn't help here either.

File: cmd/airplane/tasks/dev/server.go [hunk 2]
```diff
@@ -264,6 +265,17 @@ func runLocalDevServer(ctx context.Context, cfg taskDevConfig) error {
 		logger.Log("")
 		go func() {
 			fmt.Scanln()
```
Comment (colinking) [inline cmd/airplane/tasks/dev/server.go:267]: Can you inspect the output of `fmt.Scanln` to check if it's a ctrl-c? 

Based on https://pkg.go.dev/fmt#Scanln -- you might be able to tell this based on one of the return values, or by passing in an arg to get whatever they typed in.

**Review by colinking — CHANGES_REQUESTED** [review]
Not against this, but curious if there's a better approach.

Also your PR is still in draft, did you mean that?

File: cmd/airplane/tasks/dev/server.go (hunk 2 above)
Comment (calufornia) [reply cmd/airplane/tasks/dev/server.go:267]: Yep, I've done that—it's nothing. If you print out how many bytes are written, it's 0—no error either

Comment (colinking) [reply cmd/airplane/tasks/dev/server.go:267]: You mentioned `bufio.NewScanner`, but have you tried `bufio.NewReader().ReadBytes('\n')`? Based on the docs https://pkg.go.dev/bufio#Reader.ReadBytes, it sounds like you'll get an error back if the bytes do not end in a newline (which I assume will be the case when ctrl-c).

Comment (calufornia) [reply cmd/airplane/tasks/dev/server.go:267]: 😮 I think you might be on to something—`ReadBytes()` seems to return an error when it encounter ctrl+c

Comment (calufornia) [reply cmd/airplane/tasks/dev/server.go]: Updated with the new approach—thanks!

File: cmd/airplane/tasks/dev/server.go [hunk 3]
```diff
@@ -263,7 +264,14 @@ func runLocalDevServer(ctx context.Context, cfg taskDevConfig) error {
 		logger.Log("Press ENTER to open the studio in the browser.")
 		logger.Log("")
 		go func() {
-			fmt.Scanln()
+			// Use bufio.NewReader to properly handle interrupts on Windows. fmt.Scanln() does not return an error
+			// when interrupted on Windows, and so there's no way to distinguish between a user pressing enter vs
+			// Ctrl+C here.
```
Comment (colinking) [inline cmd/airplane/tasks/dev/server.go:269]: FYI there's a comment above about `fmt.Scanln` that should be tweaked.

**Review by colinking — APPROVED** [review]
Nice!

**Pull request merged**